x/ccv/provider/keeper/staking_keeper_interface.go

```diff
@@ -0,0 +1,80 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// IterateBondedValidatorsByPower iterates over the consensus-active validators by power.
+// The same as IterateBondedValidatorsByPower in the StakingKeeper,
+// but only returns the first MaxProviderConsensusValidators validators.
+// This is used to implement the interface of the staking keeper to interface with
+// modules that need to reference the consensus-active validators.
+func (k Keeper) IterateBondedValidatorsByPower(ctx context.Context, fn func(index int64, validator stakingtypes.ValidatorI) (stop bool)) error {
+	maxProviderConsensusVals := k.GetMaxProviderConsensusValidators(sdk.UnwrapSDKContext(ctx))
+	counter := int64(0)
+	return k.stakingKeeper.IterateBondedValidatorsByPower(ctx, func(index int64, validator stakingtypes.ValidatorI) (stop bool) {
+		if counter >= maxProviderConsensusVals {
+			return true
+		}
+		counter++
+		return fn(index, validator)
+	})
+}
+
+// TotalBondedTokens gets the amount of tokens of the consensus-active validators.
+// The same as TotalBondedTokens in the StakingKeeper, but only counts bonded tokens
+// of the first MaxProviderConsensusValidators bonded validators.
+// This is used to implement the interface of the staking keeper to interface with
+// modules that need to reference the consensus-active validators.
+func (k Keeper) TotalBondedTokens(ctx context.Context) (math.Int, error) {
+	// iterate through the bonded validators
+	totalBondedTokens := math.ZeroInt()
+
+	k.IterateBondedValidatorsByPower(ctx, func(_ int64, validator stakingtypes.ValidatorI) (stop bool) {
+		tokens := validator.GetBondedTokens()
+		totalBondedTokens = totalBondedTokens.Add(tokens)
+		return false
+	})
+
+	return totalBondedTokens, nil
+}
+
+// IterateDelegations is the same as IterateDelegations in the StakingKeeper.
+// Necessary to implement the interface of the staking keeper to interface with
+// other modules.
+func (k Keeper) IterateDelegations(ctx context.Context, delegator sdk.AccAddress, fn func(index int64, delegation stakingtypes.DelegationI) (stop bool)) error {
+	return k.stakingKeeper.IterateDelegations(ctx, delegator, fn)
```

Comment:
Could this iterate through delegations of non-consensus-active validators here and below?
Could this be problematic?

Reply:
This is on purpose, afaict the SDK also iterates over non-active delegations